src/jetstream_kv.rs

```diff
@@ -0,0 +1,906 @@
+        self.context.publish_message(&message)?;
+
+        Ok(())
+    }
+
+    /// Remove any entries associated with the key and all historical revisions.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use nats::jetstream::KeyValueConfig;
+    /// # fn main() -> std::io::Result<()> {
+    /// # let client = nats::connect("demo.nats.io")?;
+    /// # let context = nats::jetstream::new(client);
+    /// #
+    /// # let bucket = context.create_key_value(&KeyValueConfig {
+    /// #  bucket: "purge".to_string(),
+    /// #  ..Default::default()
+    /// # })?;
+    /// #
+    /// bucket.create("foo", b"bar")?;
+    /// bucket.purge("foo")?;
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn purge(&self, key: &str) -> io::Result<()> {
+        if !is_valid_key(key) {
+            return Err(io::Error::new(io::ErrorKind::InvalidInput, "invalid key"));
+        }
+
+        let mut subject = String::new();
+        subject.push_str(&self.prefix);
+        subject.push_str(key);
+
+        let mut headers = HeaderMap::default();
+        let purge_entry = headers
+            .inner
+            .entry(KV_OPERATION.to_string())
+            .or_insert_with(HashSet::default);
+
+        purge_entry.insert(KV_OPERATION_PURGE.to_string());
+
+        let rollup_entry = headers
+            .inner
+            .entry(NATS_ROLLUP.to_string())
+            .or_insert_with(HashSet::default);
+
+        rollup_entry.insert(ROLLUP_SUBJECT.to_string());
+
+        let message = Message::new(&subject, None, b"", Some(headers));
+        self.context.publish_message(&message)?;
+
+        Ok(())
+    }
+
+    /// Returns an iterator which iterate over all the current keys.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use nats::jetstream::KeyValueConfig;
+    /// # fn main() -> std::io::Result<()> {
+    /// # let client = nats::connect("demo.nats.io")?;
+    /// # let context = nats::jetstream::new(client);
+    /// #
+    /// # let bucket = context.create_key_value(&KeyValueConfig {
+    /// #  bucket: "keys".to_string(),
+    /// #  ..Default::default()
+    /// # })?;
+    /// #
+    /// bucket.put("foo", b"fizz")?;
+    /// bucket.put("bar", b"buzz")?;
+    ///
+    /// let mut keys = bucket.keys()?;
+    ///
+    /// assert_eq!(keys.next(), Some("foo".to_string()));
+    /// assert_eq!(keys.next(), Some("bar".to_string()));
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn keys(&self) -> io::Result<Keys> {
+        let mut subject = String::new();
+        subject.push_str(&self.prefix);
+        subject.push_str(ALL_KEYS);
+
+        let subscription = self.context.subscribe_with_options(
+            &subject,
+            &SubscribeOptions::ordered()
+                .headers_only()
+                .deliver_last_per_subject()
+                .enable_flow_control()
+                .idle_heartbeat(Duration::from_millis(5000)),
```

Comment:
That heartbeat value is consistent across many methods. Should be refactored into const.

Reply:
Yeah it's also arbitrary, think I derived the value from the JavaScript implementation but lets just have the `subscribe` call default it.